src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1002,7 +1002,7 @@ public virtual RedirectToPageResult RedirectToPagePermanent(string pageName, obj
         /// <returns>The <see cref="RedirectToPageResult"/> with <see cref="RedirectToPageResult.Permanent"/> set.</returns>
         [NonAction]
         public virtual RedirectToPageResult RedirectToPagePermanent(string pageName, string pageHandler)
-            => RedirectToPagePermanent(pageName, routeValues: null);
+            => RedirectToPagePermanent(pageName, pageHandler, routeValues: null, fragment: null);
```

Comment:
This is the kind of bugs you get when you use `=>` with line breaks. You deserve this